### PR TITLE
Tox tests on python 3.7 & 3.8 #ALTOS-24050

### PR DESCRIPTION
**Pull request by nnegri**
Run tox tests on python 3.7 and 3.8, with one code change.
I was having issues trying to install 3.9 so just keeping it to these for now.
Backend update: AltSchool/vishnu-backend#7217

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -64,3 +64,7 @@ lint: clean_working_directory
 format: clean_working_directory
 	$(call header,"Auto-formatting code")
 	@find $(APP_NAME) -type f -name '*.py' | xargs $(INSTALL_DIR)/bin/flake8 | sed -E 's/^([^:]*\.py).*/\1/g' | uniq | xargs autopep8 --experimental -a --in-place
+
+tox:
```
Comment (nnegri) [inline Makefile:68]: Copied this command from django-s3-assets

File: dynamic_rest_client/query.py [hunk 2]
```diff
@@ -167,7 +167,7 @@ def __iter__(self):
                 if self._page == self._pages:
                     # end of results
                     self._reset()
-                    raise StopIteration()
+                    return
```
Comment (nnegri) [inline dynamic_rest_client/query.py:170]: Necessary for new 3.7 behavior

> PEP 479 is enabled for all code in Python 3.7, meaning that StopIteration exceptions raised directly or indirectly in coroutines and generators are transformed into RuntimeError exceptions. (Contributed by Yury Selivanov in bpo-32670.)

https://stackoverflow.com/questions/51700960/runtimeerror-generator-raised-stopiteration-every-time-i-try-to-run-app

File: tox.ini [hunk 3]
```diff
@@ -2,7 +2,7 @@
 addopts=--tb=short
 
 [tox]
-envlist = py27, py36
+envlist = py36, py37, py38
```
Comment (nnegri) [inline tox.ini:5]: Do we still need to test against 2.7?

Comment (suavesav) [reply tox.ini:5]: nope!

**Review by suavesav — APPROVED** [review]
looks good! would also be good to do a package version bump in here

File: dynamic_rest_client/query.py (hunk 2 above)
Comment (suavesav) [reply dynamic_rest_client/query.py:170]: interesting, should be careful to test vishnu-backend on a feature branch when updating this repo there

**Pull request merged**